File: src/backend/Intelibill.Application/Features/Discounts/DTOs/DiscountRuleListItemDto.cs
using Intelibill.Domain.Enums;

namespace Intelibill.Application.Features.Discounts.DTOs;

public sealed record DiscountRuleListItemDto(
    Guid Id,
    DiscountRuleType RuleType,
    string Name,
    bool IsActive,
    DateTimeOffset? StartsAt,
    DateTimeOffset? EndsAt,
    DateTimeOffset CreatedAt);

File: src/mobile/android/intelibill_mobile/lib/src/features/discounts/data/dto/discount_rule_list_item_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'discount_rule_list_item_dto.freezed.dart';
part 'discount_rule_list_item_dto.g.dart';

@freezed
sealed class DiscountRuleListItemDto with _$DiscountRuleListItemDto {
  const factory DiscountRuleListItemDto({
    @JsonKey(name: 'id') required String id,
    @JsonKey(name: 'ruleType') required String ruleType,
    @JsonKey(name: 'name') required String name,
    @JsonKey(name: 'isActive') required bool isActive,
    @JsonKey(name: 'startsAt') DateTime? startsAt,
    @JsonKey(name: 'endsAt') DateTime? endsAt,
    @JsonKey(name: 'createdAt') required DateTime createdAt,
  }) = _DiscountRuleListItemDto;

  factory DiscountRuleListItemDto.fromJson(Map<String, dynamic> json) =>
      _$DiscountRuleListItemDtoFromJson(json);
}
